pkg/gofr/datasource/dgraph/dgraph_test.go

```diff
@@ -0,0 +1,262 @@
+package dgraph
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/dgraph-io/dgo/v210/protos/api"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func setupDB(t *testing.T) (*Client, *MockDgraphClient, *MockLogger, *MockMetrics) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockLogger := NewMockLogger(ctrl)
+	mockMetrics := NewMockMetrics(ctrl)
+
+	config := Config{Host: "localhost", Port: "9080"}
+	client := New(config)
+	client.UseLogger(mockLogger)
+	client.UseMetrics(mockMetrics)
+
+	mockDgraphClient := NewMockDgraphClient(ctrl)
+	client.client = mockDgraphClient
+
+	return client, mockDgraphClient, mockLogger, mockMetrics
+}
+
+func TestClient_Connect_Success(t *testing.T) {
+	client, _, mockLogger, mockMetrics := setupDB(t)
+
+	mockLogger.EXPECT().Logf(gomock.Any(), gomock.Any()).Times(2)
+
+	// Mock Metric behavior
+	mockMetrics.EXPECT().NewHistogram(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+
+	// Perform the connect operation
+	client.Connect()
+
+	require.True(t, mockLogger.ctrl.Satisfied())
+	require.True(t, mockMetrics.ctrl.Satisfied())
+}
+
+func Test_Query_Success1(t *testing.T) {
```

Comment:
we can have a better name here - just having 1 doesn't help us with the use case this test is for

Reply:
changed name to `Test_Query_Success`